sdk/communication/Azure.Communication.Common/tests/CommunicationTestEnvironment.cs

```diff
@@ -39,5 +40,10 @@ public class CommunicationTestEnvironment : TestEnvironment
         public bool ShouldIgnoreSMSTests => bool.Parse(SkipSmsTest);
 
         public bool ShouldIgnorePhoneNumbersTests => bool.Parse(SkipPhoneNumbersTest);
+
+        public string CommunicationPhoneNumberId => GetOptionalVariable(AzurePhoneNumberId);
+        public string CommunicationTestPhoneNumber1 => GetRecordedOptionalVariable("AZURE_PHONE_NUMBER_1", options => options.IsSecret("+14255550123"));
+        public string CommunicationTestPhoneNumber2 => GetRecordedOptionalVariable("AZURE_PHONE_NUMBER_2", options => options.IsSecret("+14255550123"));
+        public string CommunicationTestPhoneNumber3 => GetRecordedOptionalVariable("AZURE_PHONE_NUMBER_3", options => options.IsSecret("+14255550123"));
```

Comment:
This doesn't seem scalable. Let's assume we want to expand our test matrix (new OS, new language, etc.). We'd need to go back here and add new variables. Wouldn't be possible to set the phone numbers in the Keyvault by the agent test name instead of an integer Id?

Also, I'd rather to define the phone numbers by the agent test name, e.g. "macOS-10.15_NETCore3.", "macOS-10.15_NET6.0", etc. because repeating Ids here would be easy, and then the tests would start failing again.

Reply:
That's a great suggestion, I will take a look into it!

Reply:
@lucasrsant can you please take a look at the changes introduced in https://github.com/Azure/azure-sdk-for-net/pull/26910/commits/eed211b6ffccd4d26d2478419e702953de969138 ?